### PR TITLE
Add commit SHA + GitHub URL to SARIF output

### PR DESCRIPTION
**Pull request by shati-patel**
Add information to [`versionControlProvenance`](https://docs.oasis-open.org/sarif/sarif/v2.1.0/csprd01/sarif-v2.1.0-csprd01.html#_Ref511829897) in the SARIF output.

### CODE REVIEW

File: src/codeql.ts [hunk 1]
```diff
@@ -104,6 +104,8 @@ async function runQuery(
     outputSarif(
       codeql,
       bqrs,
+      nwo,
+      dbMetadata.creationMetadata?.sha || "HEAD",
```
Comment (rneatherway) [inline src/codeql.ts:108]: Could we pass this in as a `string?` instead and then not write the `revisionId` field if it isn't available? I think that would be a little nicer than writing `HEAD`. The extension could then take the decision to use `HEAD` in constructing the links if it doesn't find `revisionId` in the downloaded SARIF file.

**Review by rneatherway — COMMENTED** [review]
One small comment. It'd also be worth trying the produced SARIF file in https://sarifweb.azurewebsites.net/ as an extra safety net.

File: src/codeql.ts (hunk 1 above)
Comment (shati-patel) [reply src/codeql.ts]: Thanks! Attempted to do this in https://github.com/dsp-testing/qc-run2/pull/655/commits/4b53aeeaeca3dbdbf8acd4a0bb09f9432c46d4e5 🙇🏽‍♀️

**Review by rneatherway — APPROVED** [review]
:ship: :tada:

**Pull request merged**